### PR TITLE
Add `searchsorted` support

### LINKED ISSUE #627: Support `searchsorted` to search the position of a number on a sorted 1-d tensor
**Is your feature request related to a problem? Please describe.**

Given a sorted 1-d tensor and a number, `searchsorted` could be used to search the position where the number could be put into.

**Describe the solution you'd like**

For numpy, `searchsorted` is implemented by bisect search, but for the parallel situation,  perform the calculation in a sequential way sounds unacceptable.

Thus I want to trigger `numpy.searchsorted` on each chunk, record the data and position, then merge the intermediate data together to calculate the final position.

### PR DESCRIPTION
**Pull request by qinxuye**
## What do these changes do?

This PR implements `numpy.searchsorted` operation for Mars tensor. 

~~One more thing to do is adding order support for `searchsorted`, that would happen once #595 got merged.~~

- [x] Support order.

## Related issue number

Resolves #627

### CODE REVIEW

File: mars/tensor/base/searchsorted.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+            for idx in itertools.product(*(range(s) for s in v.shape)):
+                ind = xp.searchsorted(inp_data[(slice(None),) + idx], v[idx], side=op.side)
+                if ind >= len(inp_indices):
+                    ind -= 1
+                ret_indices[idx] = inp_indices[(ind,) + idx]
+                ret_data[idx] = inp_data[(ind,) + idx]
+            return ret_indices, ret_data
+
+    @classmethod
+    def _execute_reduce(cls, xp, a, v, op):
```
Comment (hekaisheng) [inline mars/tensor/base/searchsorted.py]: This method seems almost same as `_execute_combine`, should they share the same code ?

Comment (qinxuye) [reply mars/tensor/base/searchsorted.py]: It's hard to extract common part.

File: mars/tensor/base/searchsorted.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            inputs.append(v)
+            shape = v.shape
+        else:
+            shape = ()
+        return self.new_tensor(inputs, shape=shape, order=TensorOrder.C_ORDER)
+
+    @classmethod
+    def _tile_one_chunk(cls, op, a, v, out):
+        chunk_op = op.copy().reset_key()
+        in_chunks = [a.chunks[0]]
```
[the remark this answers is not in the document]
Comment (qinxuye) [reply mars/tensor/base/searchsorted.py]: I need a new list to operate, didn't know what gonna happen if the operation would modify the original list.

File: mars/tensor/base/searchsorted.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    @classmethod
+    def tile(cls, op):
+        a = op.inputs[0]
+        out = op.outputs[0]
+        input_len = len(op.inputs)
+        if input_len == 1:
+            v = op.values
+        else:
+            v = op.inputs[1]
+            v = v.reshape(v.shape).single_tiles()
```
[the remark this answers is not in the document]
Comment (qinxuye) [reply mars/tensor/base/searchsorted.py]: Mistake, I attended to do `rechunk`, but with the new commit, rechunk is not required either.

**Comment by wjsi** [comment]
Seems that cases with multiple inputs are not tested?

**Comment by qinxuye** [comment]
> Seems that cases with multiple inputs are not tested?

True, fixed already.

File: mars/tensor/base/searchsorted.py [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+    @classmethod
+    def _execute_without_stage(cls, xp, a, v, op):
+        return xp.searchsorted(a, v, side=op.side)
+
+    @classmethod
+    def _execute_map(cls, xp, a, v, op):
+        indices = xp.atleast_1d(xp.searchsorted(a, v, side=op.side))
+        data_indices = indices.copy()
+        data_indices = xp.subtract(data_indices, 1, out=data_indices,
```
Comment (wjsi) [inline mars/tensor/base/searchsorted.py]: Add a comment for this behavior

**Pull request merged**